### PR TITLE
Fix gift card emails + add logging for email failures

### PR DESCRIPTION
**Pull request by juliexxia**

### CODE REVIEW

File: app/controllers/webhooks_controller.rb [hunk 1]
```diff
@@ -149,6 +149,10 @@ def handle_payment_intent_succeeded(
           )
         rescue StandardError
           # don't let a failed email bring down the whole POST
+          logger.error "Pool donation email errored out. "
+          + "email: #{payment_intent.recipient.email}, "
+          + "receipt: #{payment_intent.receipt_url} "
+          + "amount: #{amount}, "
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:155]: nit shouldn't have a trailing comma

File: app/controllers/webhooks_controller.rb [hunk 2]
```diff
@@ -149,6 +149,10 @@ def handle_payment_intent_succeeded(
           )
         rescue StandardError
           # don't let a failed email bring down the whole POST
+          logger.error "Pool donation email errored out. "
+          + "email: #{payment_intent.recipient.email}, "
+          + "receipt: #{payment_intent.receipt_url} "
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:154]: nit should have a trailing comma

File: app/controllers/webhooks_controller.rb [hunk 3]
```diff
@@ -168,6 +172,11 @@ def handle_payment_intent_succeeded(
             )
           rescue StandardError
             # don't let a failed email bring down the whole POST
+            logger.error "Donation email errored out. "
+            + "email: #{payment_intent.recipient.email}, "
+            + "receipt: #{payment_intent.receipt_url} "
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:177]: nit should have a trailing comma lol

File: app/controllers/webhooks_controller.rb [hunk 4]
```diff
@@ -168,6 +172,11 @@ def handle_payment_intent_succeeded(
...
+            + "amount: #{amount}, "
+            + "merchant: #{merchant_name}, "
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:179]: lol

File: app/controllers/webhooks_controller.rb [hunk 5]
```diff
@@ -191,7 +200,14 @@ def handle_payment_intent_succeeded(
               receipt_id: gift_card_detail.seller_gift_card_id
             )
           rescue StandardError
-            logger.info "Gift card email errored out. payment_intent: #{payment_intent}, amount: #{amount}, merchant: #{merchant_name}, receipt_id: #{gift_card_detail.seller_gift_card_id}, gift card detail: #{gift_card_detail}"
+            # don't let a failed email bring down the whole POST
+            logger.error "Gift card email errored out. "
+            + "email: #{payment_intent.recipient.email}, "
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:205]: nit this should be tabbed right two spaces — maybe run rubocop -a?

File: app/controllers/webhooks_controller.rb [hunk 6]
```diff
@@ -337,7 +352,7 @@ def send_gift_card_receipt(payment_intent:, amount:, merchant:, receipt_id:)
            '<p> the Send Chinatown Love team</p>' \
            '</body>' \
            '</html>'
-    send_receipt(to: payment_intent.email, html: html)
+    send_receipt(to: payment_intent.recipient.email, html: html)
```
Comment (jtmckibb) [inline app/controllers/webhooks_controller.rb:355]: LOL omg

Comment (jtmckibb) [reply app/controllers/webhooks_controller.rb:355]: maybe we can do a logger.info("sending email to ...") in the send_receipt method? 

ideally we'd abstract the emailed receipt sender out into a service or a helper (idk the difference in Rails)

**Comment by juliexxia** [comment]
Did some rubocopping/manual formatting

**Pull request merged**